datumaro/plugins/sampler/random_sampler.py

```diff
@@ -83,3 +87,117 @@ def __iter__(self):
                             return
 
                 i += 1
+
+class LabelRandomSampler(Transform, CliPlugin):
+    """
+    Sampler that keeps at least the required number of annotations of
+    each class in the dataset for each subset separately.|n
+    |n
+    Consider using the "stats" command to get class distribution in
+    the dataset.|n
+    |n
+    Notes:|n
+    - Items can contain annotations of several selected classes
+    (e.g. 3 bounding boxes per image). The number of annotations in the
+    resulting dataset varies between max(class counts) and sum(class counts)|n
+    - If the input dataset does not has enough class annotations, the result
+    will contain only what is available|n
+    - Items are selected uniformly|n
+    - For reasons above, the resulting class distribution in the dataset may
+    not be the same as requested|n
+    - The resulting dataset will only keep annotations for
+    classes with specified count > 0|n
+    |n
+    Example: select at least 5 annotations of each class randomly|n
+    |s|s%(prog)s -k 5 |n
+    Example: select at least 5 images with "cat" annotations and 3 "person"|n
+    |s|s%(prog)s -l "cat:5" -l "person:3" |n
+    """
+
+    @staticmethod
+    def _parse_label_count(s: str) -> Tuple[str, int]:
+        label, count = s.split(':', maxsplit=1)
+        count = cast(count, int, default=None)
+
+        if not label:
+            raise argparse.ArgumentError(None, "Class name cannot be empty")
+        if count is None or count < 0:
+            raise argparse.ArgumentError(None,
+                f"Class '{label}' count is invalid")
+
+        return label, count
+
+    @classmethod
+    def build_cmdline_parser(cls, **kwargs):
+        parser = super().build_cmdline_parser(**kwargs)
+        parser.add_argument('-k', '--count', type=int, required=True,
+            help="Minimum number of annotations of each class")
```

Comment:
As I understand it, this is the minimum number of elements not for each class, but for classes that are not specified with the `label_counts` argument.

Reply:
Yes, it is.